src/events/math.js

```diff
@@ -0,0 +1,98 @@
+import calculate from './calculate';
+
+function isNum(item) {
+  return !!item.match(/[0-9]+/);
+}
+
+export default function maths(ob, btnName) {
+  if (btnName === 'AC') {
+    return {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+  }
+
+  if (isNum(btnName)) {
+    if (btnName === '0' && ob.next === '0') {
+      return {};
+    }
+    if (ob.calculation) {
+      if (ob.next) {
+        return { ...ob, next: ob.next + btnName };
+      }
+      return { ...ob, next: btnName };
+    }
+
+    if (ob.next) {
+      return {
+        next: ob.next + btnName,
+        total: null,
+      };
+    }
+    return {
+      next: btnName,
+      total: null,
+    };
+  }
+
+  if (btnName === '.') {
+    if (ob.next) {
+      if (ob.next.includes('.')) {
+        return { ...ob };
+      }
+      return { ...ob, next: `${ob.next}` };
+    }
+    if (ob.total) {
+      if (ob.total.includes('.')) {
+        return {};
+      }
+      return { total: `${ob.total}.` };
+    }
+    return { total: '0.' };
+  }
+
+  if (btnName === '=') {
+    if (ob.next && ob.calculation) {
+      return {
+        total: calculate(ob.total, ob.next, ob.calculation),
+        next: null,
+        calculation: null,
+      };
```

Comment:
- [ ] Another observation made was when I tried to carry out a divide operation and then I press `=`, I get a `Not divisible by 0.` as the `result/quotient` instead of the real result. For example, when I do `9/3` and press the `=` button, I get a `Not divisible by 0.` instead of `3`. It would be nice to fix it also.